provider/coredns.go

```diff
@@ -258,6 +258,28 @@ func NewCoreDNSProvider(domainFilter endpoint.DomainFilter, prefix string, dryRu
 	}, nil
 }
 
+// Find takes a Endpoint slice and looks for an element in it. If found it will
+// return it's key, otherwise it will return -1 and a bool of false.
+func findEp(slice []*endpoint.Endpoint, dnsName string) (int, bool) {
```

Comment:
Given that you only use the `Endpoint` in https://github.com/kubernetes-sigs/external-dns/pull/1475/files#diff-5e7d0936cf66e74a78954bf63c9a428aR301-R302, what about making this return directly the `Endpoint` instead of the index and simplify https://github.com/kubernetes-sigs/external-dns/pull/1475/files#diff-5e7d0936cf66e74a78954bf63c9a428aR301-R303 ? 

An alternative is to ditch this function completely and embed it in the `Records` function as it is very small.

Reply:
@Raffo very good idea, I've implemented it in https://github.com/kubernetes-sigs/external-dns/pull/1475/commits/c9f22b4b3bd0bae55e95b450095dfb0294885a8d .